### PR TITLE
Use tax rate ids

### PR DESCRIPTION
**Pull request by jenssegers**

### CODE REVIEW

**Review by jdrieghe — APPROVED** [review]
Looks legit

**Review by stivni — DISMISSED** [review]
I don't know if we throw too much of the immutability measures we took overboard... please rediscuss with me

File: apiary.apib [hunk 1]
```diff
@@ -475,29 +477,32 @@ Get details for a single invoice.
...
                 + tax_inclusive (Money)
                 + payable (Money)
                 + taxes (array)
                     + (object)
-                        + rate: `0.21` (number)
-                        + taxable (Money)
-                        + tax (Money)
+                        + tax (object)
+                            + type: `taxRate`
+                            + id: `c0c03f1e-77e3-402c-a713-30ea1c585823` (string)
```
Comment (luisabritotl) [inline apiary.apib]: Why do we need tax (type, id) in each of the line items and also in the total? Is is a default tax for the invoice?

**Comment by jdrieghe** [comment]
@stivni @jenssegers if you'd like us to change something by tomorrow, we're going to need a conclusion here fairly soon.

**Comment by jenssegers** [comment]
@stivni I checked with invoicing team, and tax rates are not adjustable. So there should not be too many immutability issues right?

File: apiary.apib [hunk 2]
```diff
@@ -425,9 +425,11 @@ Get a list of invoices.
                     + payable (Money)
                     + taxes (array)
                         + (object)
-                            + rate: `0.21` (number)
-                            + taxable (Money)
-                            + tax (Money)
+                            + tax (object)
```
Comment (jenssegers) [inline apiary.apib]: rate?

**Review by jdrieghe — APPROVED** [review]

**Review by steftrenson — APPROVED** [review]

**Review by jdrieghe — APPROVED** [review]
:rocket:

**Review by luisaterrabrito — APPROVED** [review]

**Comment by jdrieghe** [comment]
@jenssegers or @steftrenson can you fix conflicts here so we can merge this?

**Pull request merged**